lib/shopify_api/webhooks/registration.rb

```diff
@@ -48,8 +48,28 @@ def mutation_name(webhook_id); end
       sig { abstract.returns(String) }
       def build_check_query; end
 
-      sig { abstract.params(body: T::Hash[String, T.untyped]).returns(T::Hash[Symbol, String]) }
-      def parse_check_result(body); end
+      sig do
+        params(body: T::Hash[String, T.untyped]).returns({
+          webhook_id: T.nilable(String),
+          current_address: T.nilable(String),
+          fields: T::Array[String],
+          metafield_namespaces: T::Array[String],
+        })
+      end
+      def parse_check_result(body)
```

Comment:
Is this the "ruby way" to remove duplicate code i.e. with a superclass & subclass implementation. Is there a good way to ensure this never gets called by itself?